File: xadapterLibrary/src/main/java/com/xadapter/widget/SimpleRefresh.java
package com.xadapter.widget;

import android.content.Context;
import android.graphics.drawable.AnimationDrawable;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatImageView;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.animation.Animation;
import android.view.animation.RotateAnimation;

import com.xadapter.R;

/**
 * by y on 2016/11/16
 */

public class SimpleRefresh extends XRefreshView {

    private AppCompatImageView progressBar;
    private AppCompatTextView mText;
    private AnimationDrawable animationDrawable;
    private AppCompatImageView mTipsIv;

    private Animation mRotateUpAnim;
    private Animation mRotateDownAnim;

    public SimpleRefresh(Context context) {
        super(context);
    }

    public SimpleRefresh(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public SimpleRefresh(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void initView() {
        progressBar = (AppCompatImageView) findViewById(R.id.progressbar);
        mText = (AppCompatTextView) findViewById(R.id.tv_tips);
        mTipsIv = (AppCompatImageView) findViewById(R.id.iv_tips);
        animationDrawable = (AnimationDrawable) progressBar.getBackground();
        mText.setText("下拉刷新");
        initAnimation();
    }

    private void initAnimation() {
        mRotateUpAnim = new RotateAnimation(0.0f, -180.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        mRotateUpAnim.setDuration(180);
        mRotateUpAnim.setFillAfter(true);
        mRotateDownAnim = new RotateAnimation(-180.0f, 0.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        mRotateDownAnim.setDuration(180);
        mRotateDownAnim.setFillAfter(true);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.simple_refresh;
    }

    @Override
    protected void onStart() {
        mTipsIv.clearAnimation();
        mTipsIv.setVisibility(VISIBLE);
        progressBar.setVisibility(INVISIBLE);
    }

    @Override
    protected void onNormal() {
        if (mState == READY) {
            mTipsIv.startAnimation(mRotateDownAnim);
        } else {
            mTipsIv.clearAnimation();
        }
        mTipsIv.setVisibility(VISIBLE);
        progressBar.setVisibility(INVISIBLE);
        animationDrawable.stop();
        mText.setText("下拉刷新");
    }

    @Override
    protected void onReady() {
        mTipsIv.startAnimation(mRotateUpAnim);
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(VISIBLE);
        animationDrawable.stop();
        mText.setText("释放立即刷新");
    }

    @Override
    protected void onRefresh() {
        progressBar.setVisibility(VISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        animationDrawable.start();
        mText.setText("正在刷新...");
    }

    @Override
    protected void onSuccess() {
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        animationDrawable.stop();
        mTipsIv.clearAnimation();
        mText.setText("刷新成功");
    }

    @Override
    protected void onError() {
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        animationDrawable.stop();
        mTipsIv.clearAnimation();
        mText.setText("刷新失败");
    }


}

File: xadapterLibrary/src/main/java/com/xadapter/simple/SimpleRefresh.kt
package com.xadapter.simple

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.AnimationDrawable
import android.util.AttributeSet
import android.view.View
import android.view.animation.Animation
import android.view.animation.RotateAnimation
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView

import com.xadapter.R
import com.xadapter.widget.XRefreshView

/**
 * by y on 2016/11/16
 */

class SimpleRefresh : XRefreshView {

    private lateinit var progressBar: AppCompatImageView
    private lateinit var mText: AppCompatTextView
    private lateinit var animationDrawable: AnimationDrawable
    private lateinit var mTipsIv: AppCompatImageView

    private lateinit var mRotateUpAnim: Animation
    private lateinit var mRotateDownAnim: Animation

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    public override fun initView() {
        progressBar = findViewById(R.id.progressbar)
        mText = findViewById(R.id.tv_tips)
        mTipsIv = findViewById(R.id.iv_tips)
        animationDrawable = progressBar.background as AnimationDrawable
        mText.text = "下拉刷新"
        mText.setTextColor(Color.BLACK)
        initAnimation()
    }

    private fun initAnimation() {
        mRotateUpAnim = RotateAnimation(0.0f, -180.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        mRotateUpAnim.duration = 180
        mRotateUpAnim.fillAfter = true
        mRotateDownAnim = RotateAnimation(-180.0f, 0.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        mRotateDownAnim.duration = 180
        mRotateDownAnim.fillAfter = true
    }

    override fun getLayoutId(): Int {
        return R.layout.simple_refresh
    }

    override fun onStart() {
        mTipsIv.clearAnimation()
        mTipsIv.visibility = View.VISIBLE
        progressBar.visibility = View.INVISIBLE
    }

    override fun onNormal() {
        if (state == READY) {
            mTipsIv.startAnimation(mRotateDownAnim)
        } else {
            mTipsIv.clearAnimation()
        }
        mTipsIv.visibility = View.VISIBLE
        progressBar.visibility = View.INVISIBLE
        animationDrawable.stop()
        mText.text = "下拉刷新"
    }

    override fun onReady() {
        mTipsIv.startAnimation(mRotateUpAnim)
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.VISIBLE
        animationDrawable.stop()
        mText.text = "释放立即刷新"
    }

    override fun onRefresh() {
        progressBar.visibility = View.VISIBLE
        mTipsIv.visibility = View.INVISIBLE
        animationDrawable.start()
        mText.text = "正在刷新..."
    }

    override fun onSuccess() {
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.INVISIBLE
        animationDrawable.stop()
        mTipsIv.clearAnimation()
        mText.text = "刷新成功"
    }

    override fun onError() {
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.INVISIBLE
        animationDrawable.stop()
        mTipsIv.clearAnimation()
        mText.text = "刷新失败"
    }

}
